src/scancode/api.py

```diff
@@ -105,30 +105,36 @@ def get_emails(location):
     """
     Yield mappings of emails detected in the file at `location`.
     """
+    email_threshold  = 50
     from cluecode.finder import find_emails
-    for email, line_num  in find_emails(location):
+    for email_count, (email, line_num)  in enumerate(find_emails(location)):
         if not email:
             continue
         misc = OrderedDict()
         misc['email'] = email
         misc['start_line'] = line_num
         misc['end_line'] = line_num
         yield misc
+        if email_count >= email_threshold:
+            break
 
 
 def get_urls(location):
     """
     Yield mappings of urls detected in the file at `location`.
     """
+    url_threshold  = 50 
     from cluecode.finder import find_urls
-    for urls, line_num  in find_urls(location):
+    for url_count, (urls, line_num)  in enumerate(find_urls(location)):
         if not urls:
             continue
         misc = OrderedDict()
         misc['url'] = urls
         misc['start_line'] = line_num
         misc['end_line'] = line_num
         yield misc
+        if url_count >= url_threshold:
+            break
```

Comment:
Using `itertools.slice` would likely be best, in which case this would take place not here but in the corresponding url and email plugins in the #885 branch?
We could accept leave this API unchanged and in `get_scanner` https://github.com/nexB/scancode-toolkit/blob/877c03e5fd54e099102253e73d0100c1f7407dd7/src/scancode/plugin_url.py#L53 return a closure with `functools.partial` that would wrap its results in an `itertools.islice based on the received argument of a new option `--max-urls`  option added in  https://github.com/nexB/scancode-toolkit/blob/877c03e5fd54e099102253e73d0100c1f7407dd7/src/scancode/plugin_url.py 

What do you think?

Reply:
@pombredanne Yes , sounds good . it'd take some time for me to look at `how to add another option to the plugin` at #787  . I tried skimming the branch , couldn't figure out from where is `-u` sets `url` ( of kwargs ) to `true` I mean from where is URLScanner getting called  .